src/styles/_catalog.scss

```diff
@@ -16,7 +16,7 @@
     margin-bottom: 48px;
   }
 
-  &__card:not(:nth-child(4n)) {
-    margin-right: 48px;
+  @media (max-width: 1024px) {
+    grid-template-columns: repeat(auto-fit, 200px);
```

Comment:
this is the shortest way to add a grid, but according to the task, you need the columns to change at a specific resolution. I recommend adding media queries for each resolution.

Reply:
media queries for each resolution would be better solution but this one also good 👍